src/CMakeLists.txt

```diff
@@ -70,6 +70,11 @@ if (MSVC)
   )
 endif (MSVC)
 
+if (ENABLE_ALL_WARNINGS)
+  target_compile_options(objxgboost PUBLIC
+    $<IF:$<COMPILE_LANGUAGE:CUDA>,-Xcompiler=-Wall -Xcompiler=-Wextra,-Wall -Wextra>)
```

Comment:
Note: NVCC doesn't accept `-Wall` option. We need to pass `-Wall` to the host compiler, so use `-Xcompiler=-Wall` instead.